src/app/components/layout/LayoutHeader.vue

```diff
@@ -1,78 +1,112 @@
-        >
-          <AccountProfilePicture
-            :account-id="store.signedInAccountId"
-            class="size-10"
-            height="40"
-            width="40"
-          />
-        </ButtonIcon>
-        <ButtonText
-          v-else
-          :aria-label="t('signIn')"
-          class="hidden 2xl:inline-block"
-          :to="localePath('session-create')"
-        >
-          {{ t('signIn') }}
-        </ButtonText>
-        <ButtonIcon
-          :aria-label="t('settings')"
-          :to="
-            localePath({
-              name: 'session-edit-id',
-              params: { id: store.jwtDecoded?.id || 'anonymous' },
-            })
-          "
-        >
-          <IHeroiconsBars3 height="2em" width="2em" />
-        </ButtonIcon>
+        <slot name="right">
+          <template v-if="!showButtons">
+            <ButtonEventNew />
+            <div
+              class="my-1 hidden w-px flex-none self-stretch bg-gray-300 2xl:flex dark:bg-gray-600"
+            />
+            <ButtonIcon
+              v-if="store.signedInAccountId"
+              :aria-label="t('dashboard')"
+              class="hidden rounded-full 2xl:block"
+              :title="t('dashboard')"
+              :to="localePath(`dashboard`)"
+            >
+              <AccountProfilePicture
+                :account-id="store.signedInAccountId"
+                classes="h-10 w-10"
+                height="40"
+                width="40"
+              />
+            </ButtonIcon>
+            <ButtonText
+              v-else
+              :aria-label="t('signIn')"
+              class="hidden 2xl:inline-block"
+              :to="localePath('session-create')"
+            >
+              {{ t('signIn') }}
+            </ButtonText>
+            <ButtonIcon
+              :aria-label="t('settings')"
+              :to="
+                localePath({
+                  name: 'session-edit-id',
+                  params: { id: store.jwtDecoded?.id || 'anonymous' },
+                })
+              "
+            >
+              <IHeroiconsBars3 height="2em" width="2em" />
+            </ButtonIcon>
+          </template>
+          <div v-else class="size-6"></div>
+        </slot>
       </div>
     </div>
-  </header>
-  <!-- </Loader> -->
+  </div>
 </template>
 
 <script setup lang="ts">
 import { isEqual } from 'ufo'
+import { useHeaderTitle } from '~/composables/useHeaderTitle'
 
+const headerTitle = useHeaderTitle()
 const store = useStore()
 const localePath = useLocalePath()
 const { t } = useI18n()
 const route = useRoute()
+
+const showButtons = computed(() => {
+  const isLoginPage =
+    String(route.path).includes('account') ||
+    String(route.path).includes('session')
+  return isLoginPage
+})
+
+const showBackButton = computed(() => {
+  const isSettings = String(route.path).includes('session/edit')
+  const isEvents = String(route.path).includes('event')
+  return isSettings || isEvents
+})
```

Comment:
Components shouldn't really know about routes, but I'll review in more detail soon.